include/aspect/vofinterface/vof_system.h

```diff
@@ -0,0 +1,148 @@
+                     const FiniteElement<dim> &vof_element,
+                     const Mapping<dim>       &mapping,
+                     const Quadrature<dim>    &quadrature,
+                     const Quadrature<dim-1>  &face_quadrature);
+          VoFSystem (const VoFSystem &data);
+
+          FEValues<dim>          finite_element_values;
+          FEFaceValues<dim>      face_finite_element_values;
+          FEFaceValues<dim>      neighbor_face_finite_element_values;
+          FESubfaceValues<dim>   subface_finite_element_values;
+
+          std::vector<types::global_dof_index>   local_dof_indices;
+
+          /**
+           * Variables describing the values of the shape functions at the
+           * quadrature points, as they are used in the advection assembly
+           * function. note that the sizes of these arrays are equal to the
+           * number of shape functions corresponding to the advected field (and
+           * not of the overall field!), and that they are also correspondingly
+           * indexed.
+           */
+          std::vector<double>         phi_field;
+          std::vector<double>         face_phi_field;
+
+          std::vector<double>         old_field_values;
+          std::vector<Tensor<1,dim> > cell_i_n_values;
+          std::vector<double>         cell_i_d_values;
+
+          std::vector<Tensor<1,dim> > face_current_velocity_values;
+          std::vector<Tensor<1,dim> > face_old_velocity_values;
+          std::vector<Tensor<1,dim> > face_old_old_velocity_values;
+          std::vector<Tensor<1,dim> > face_mesh_velocity_values;
+
+          std::vector<double>         neighbor_old_values;
+        };
+      }
+
+      namespace CopyData
+      {
+        template <int dim>
+        struct VoFSystem
+        {
+          /**
+           * Constructor.
+           * @param finite_element The element that describes the field for which we
+           *    are trying to assemble a linear system. <b>Not</b> the global finite
+           *    element.
+           */
+          VoFSystem(const FiniteElement<dim> &finite_element);
+          VoFSystem(const VoFSystem &data);
+
+          /**
+           * Local contributions to the global matrix and right hand side
+           * that correspond only to the variables listed in local_dof_indices
+           */
+          FullMatrix<double>          local_matrix;
+          Vector<double>              local_rhs;
+          /**
+           * Local contributions to the global rhs from the face terms in the
+           * discontinuous Galerkin interpretation of the VoF method.  The
+           * vector is of length GeometryInfo<dim>::max_children_per_face *
+           * GeometryInfo<dim>::faces_per_cell so as to hold one matrix for
+           * each possible face or subface of the cell.
+           **/
+          std::vector<Vector<double>>     local_f_rhs;
+          std::vector<FullMatrix<double>> local_f_matrices_ext_ext;
```

Comment:
maybe rename to `local_face_...` to make it clearer to the reader what the `f` stands for. If you use an IDE, you can do this renaming via refactoring rather than search/replace